### ISSUE TITLE
Media Library doesn't update after dragging image onto Image block

### ISSUE THREAD

**Issue by maddisondesigns** [WordPress/gutenberg#4612]
## Issue Overview
After inserting an Image block, I then dragged 'n dropped an image directly onto the block. I then inserted another image block and after clicking the _Add from Media Library_ button, and displaying the Media popup window, the previous image that I added to the page isn't displaying in the Media Library.

New image that I dragged onto Image Block
![gutenberg-imgblockdragdrop](https://user-images.githubusercontent.com/2078245/35179551-875a8a7a-fdf0-11e7-854a-db49b7f40927.png)

Newly added image not appearing in Media LIbrary
![gutenberg-imgnotinlibrary](https://user-images.githubusercontent.com/2078245/35179553-92390d22-fdf0-11e7-8329-7d32937daa1d.png)

## Steps to Reproduce (for bugs)
1. Insert Image block
2. Drag 'n drop image onto Image block
3. Insert new Image block
4. Click the _Add from Media Library_ button and try to add the same image. You can't, because it doesn't appear in the Media Library

## Expected Behavior
If I drag'n drop an image onto an Image BLock, this should get added to my Media LIbraray, and should be displayed when I try to add another image by viewing the Media Library popup

## Current Behavior
Images dragged onto Image Block aren't reflected in Media Library popup window

_Firefox Quantum 57.0.4 (64-bit)
Gutenberg 2.0.0
WordPress 4.9.2
No other plugins activated_

**Comment by karmatosed** [comment]
Just checking the steps here as can't reproduce.

I added an image via drag and drop:

![2018-01-22 at 16 53](https://user-images.githubusercontent.com/253067/35232996-d7ba85ec-ff94-11e7-9f40-ce711cbd8e51.png)

I then clicked image block and to see if it's in the media library:

![2018-01-22 at 16 52](https://user-images.githubusercontent.com/253067/35232961-bc2eea3e-ff94-11e7-93c4-9c74f0d49076.png)

**Comment by maddisondesigns** [comment]
I can consistently replicate the issue by;

- Insert 1st Image Block
- Drag image to 1st block
- Insert 2nd Image Block
- Click _Add from Media Library_ button on 2nd block (The first image you dragged will most likely show)
- Close Popup window
- Drag new image to 2nd block
- Insert 3rd Image block
- Click _Add from Media Library_ button on new block (Image that was last dragged doesn't appear)

![gutenberg-dragdropimageblock2](https://user-images.githubusercontent.com/2078245/35251058-9a5d24ae-002d-11e8-8c7f-29d576fe47a0.gif)

**Comment by karmatosed** [comment]
Ah so there were some missing steps there. I can replicate following exactly those steps thanks. It's also in Chrome, so I don't think a browser issue.

**Comment by jeffpaul** [comment]
_This ticket was mentioned in [Slack](https://make.wordpress.org/chat/) in #core-editor by jeffpaul. [View the logs](https://wordpress.slack.com/archives/C02QB2JS7/p1520532110000420)._

**Comment by jeffpaul** [comment]
_This ticket was mentioned in [Slack](https://make.wordpress.org/chat/) in #core-editor by jeffpaul. [View the logs](https://wordpress.slack.com/archives/C02QB2JS7/p1521133665000406)._

**Comment by antpb** [comment]
The [latest commit](https://github.com/WordPress/gutenberg/pull/6390/commits/7ad2dd3293e2df86006841a79c944d85d9a09009) from https://github.com/WordPress/gutenberg/pull/6390 does fix the issue. It would need to be tested and possibly optimized. From my end, it seems to be working well but there may be some edge cases we're missing.

Mentioned by @jorgefilipecosta, 

> "in order to fix the bug we need to request all images each time we open the media library (to know if new media was added in parallel) but that makes the open of the media library slower, it is a tradeoff that we should make."

@joemcgill, we will be digging into backbone things soon. Want to get this PR on your radar if you have the bandwidth to review. If you have any thoughts on why we shouldn't go the proposed route, I'd love to hear them. 

Based on the above info, Is this a performance hit we'd be willing to take and improve on later to fix the bug for 5.0? Is there another option?

**Comment by antpb** [comment]
@joemcgill I moved the approach from that PR to the `mediaUpload` function. This seems to work from my end and doesn't require us to load the collections every time we open the modal. Agree?

PR: https://github.com/WordPress/gutenberg/pull/10843

**Comment by antpb** [comment]
@afercia was able to reproduce using the upload button (thank you!) I was able to reproduce using the upload button. I've re-opened the PR above that refreshes collection when the media library is opened. I think we should go this route and improve later. I could open an issue after we merge to optimize the collections of the Media library based on the feedback provided above. https://github.com/WordPress/gutenberg/pull/10843

**Comment by designsimply** [comment]
Tested and confirmed the bug using WordPress 4.9.8 and Gutenberg 4.1.0-rc.2 with Firefox 62.0.3 on macOS 10.13.6. ([53s](https://cloudup.com/cxOOQQ6ontS))

I noticed there is a suggested improvement on #10843 so I'll take away the `Needs Testing` label for now and if you need help testing the PR once it's ready-to-go, feel free to ping me here or ping in #core-test on WP Slack to ask for testing help!

**Comment by jorgefilipecosta** [comment]
I think closing this issue will also close https://github.com/WordPress/gutenberg/issues/11567.

**Comment by swissspidy** [comment]
@jorgefilipecosta Indeed my issue sounds like a duplicate of this one.
